external/cmake/cmakelibs.cmake

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) 2014-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed in accordance with the terms specified in
+# the LICENSE file found in the root directory of this source tree.
+
+# This function takes the global properties saved by addOsqueryExtensionEx and 
+# generates a single extension executable containing all the user code
+function(generateOsqueryExtensionGroup)
+  get_property(extension_source_files GLOBAL PROPERTY OSQUERY_EXTENSION_GROUP_SOURCES)
+  if("${extension_source_files}" STREQUAL "")
+    return()
+  endif()
+
+  if(DEFINED ENV{OSQUERY_EXTENSION_GROUP_NAME} OR ENV{OSQUERY_EXTENSION_GROUP_VERSION})
+    message("ENV {OSQUERY_EXTENSION_GROUP_NAME/VERSION} has been deprecated. Please set cache variable!")
+  endif()
+
+  if(NOT OSQUERY_EXTENSION_GROUP_NAME)
+    set(OSQUERY_EXTENSION_GROUP_NAME "osquery_extension_group" CACHE STRING "Overrides osquery extension group name")
+  endif()
```

Comment:
It's not necessary to check if the cache variable is already set, if it is, it won't be overwritten (unless you explicitly pass FORCE).